File: tests/fixtures/projects/pytest-coverage/pytest_coverage/classifier.py
"""Tiny domain function with three branches, one deliberately untested.

The test suite covers the ``positive`` and ``zero`` branches; the
``negative`` branch is intentionally left uncovered so the Coverage
engine's ``missing_branches`` / ``missing_lines`` extraction has
something concrete to point at.
"""


def classify(value: int) -> str:
    if value > 0:
        return "positive"
    if value == 0:
        return "zero"
    # Intentionally unreached by the fixture's test suite.
    return "negative"

File: tests/fixtures/projects/jest-basic-coverage/src/classifier.js
// Tiny library under test with three branches, one deliberately left
// untested so the Istanbul `coverage-final.json` carries a concrete
// uncovered region. Mirrors `pytest_coverage.classifier` so the jest
// coverage fixture stays narrow: just enough surface for a per-file
// degraded coverage signal.
function classify(value) {
  if (value > 0) {
    return 'positive';
  }
  if (value === 0) {
    return 'zero';
  }
  // Intentionally unreached by the fixture's test suite.
  return 'negative';
}

module.exports = { classify };
